### PR TITLE
feat: reschedule workflow added

### LINKED ISSUE #3719: New Workflow trigger: When event is rescheduled
Add a new trigger in Workflows for when event is rescheduled. We already have a trigger in Zapier and Webhooks for that.

### PR DESCRIPTION
**Pull request by Udit-takkar**
Fixes : #3719 

1) Create workflow with trigger "when event is rescheduled"
Screenshot 2022-08-27 at 6 00 43 PM

2)Reschedule event

3)
Screenshot 2022-08-27 at 6 01 38 PM

### CODE REVIEW

File: packages/features/ee/workflows/lib/reminders/reminderScheduler.ts [hunk 1]
```diff
@@ -23,51 +23,53 @@ export const scheduleWorkflowReminders = async (
...
-                sendTo,
-                step.emailSubject || "",
-                step.reminderBody || "",
-                step.id,
-                step.template
-              );
-            }
-          });
-        }
+      if (workflowReference.workflow.steps.length === 0) return;
```
Comment (Udit-takkar) [inline packages/features/ee/workflows/lib/reminders/reminderScheduler.ts:26]: it was mentioned in handbook to prefer something like this.( ref : https://handbook.cal.com/engineering/best-practices/prefer-early-returns)

File: packages/features/ee/workflows/lib/reminders/reminderScheduler.ts [hunk 2]
```diff
@@ -23,51 +23,53 @@ export const scheduleWorkflowReminders = async (
...
+
+      const workflow = workflowReference.workflow;
+      if (
+        workflow.trigger ===
+        (WorkflowTriggerEvents.BEFORE_EVENT ||
+          WorkflowTriggerEvents.NEW_EVENT ||
+          WorkflowTriggerEvents.RESCHEDULE_EVENT)
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/reminders/reminderScheduler.ts:33]: This never returns true 
```suggestion
        workflow.trigger === WorkflowTriggerEvents.BEFORE_EVENT ||
        workflow.trigger === WorkflowTriggerEvents.NEW_EVENT ||
        workflow.trigger === WorkflowTriggerEvents.RESCHEDULE_EVENT
```

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
Thank you for your contribution 🙏  Tested it locally and works great with the suggested change I added in the comment

File: packages/features/ee/workflows/lib/reminders/reminderScheduler.ts (hunk 2 above)
Comment (Udit-takkar) [reply packages/features/ee/workflows/lib/reminders/reminderScheduler.ts]: oh yeah forgot to change here

File: packages/features/ee/workflows/lib/reminders/smsReminderManager.ts [hunk 3]
```diff
@@ -85,10 +85,12 @@ export const scheduleSMSReminder = async (
...
   if (message.length > 0 && reminderPhone) {
-    //send SMS when event is booked/cancelled
+    //send SMS when event is booked/cancelled/Reschdeuled
     if (
-      triggerEvent === WorkflowTriggerEvents.NEW_EVENT ||
-      triggerEvent === WorkflowTriggerEvents.EVENT_CANCELLED
+      triggerEvent ===
+      (WorkflowTriggerEvents.NEW_EVENT ||
+        WorkflowTriggerEvents.EVENT_CANCELLED ||
+        WorkflowTriggerEvents.RESCHEDULE_EVENT)
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/reminders/smsReminderManager.ts]: same here
```suggestion
        workflow.trigger === WorkflowTriggerEvents.BEFORE_EVENT ||
	        workflow.trigger === WorkflowTriggerEvents.NEW_EVENT ||
	        workflow.trigger === WorkflowTriggerEvents.RESCHEDULE_EVENT
```

Comment (Udit-takkar) [reply packages/features/ee/workflows/lib/reminders/smsReminderManager.ts]: Screenshot 2022-08-30 at 9 03 52 PM

@CarinaWolli  It's already updated . i think you are seeing the previous version.

**Comment by CarinaWolli** [comment]
@Udit-takkar `NEW_EVENT` and `RESCHEDULE_EVENT` both trigger when an event is booked and rescheduled. We need to check in [api/book/event](https://github.com/calcom/cal.com/blob/main/apps/web/pages/api/book/event.ts) if this is a reschedule or if a new event is booked and then only call the appropriate trigger

**Comment by Udit-takkar** [comment]
ok i'll try to will fix this

**Comment by Udit-takkar** [comment]
seem to work fine now.

https://www.loom.com/share/87f8f0490850400187611611e090bc9d

**Review by CarinaWolli — APPROVED** [review]
Great work @Udit-takkar 🙏

**Comment by CarinaWolli** [comment]
Type checks are failing now. If I am not mistaken we can just add false as the 5th argument and it should work. @Udit-takkar could you fix that please?
Screenshot 2022-08-31 at 18 37 57

**Comment by Udit-takkar** [comment]
thanks @PeerRich  for resolving the type error

**Pull request merged**